freqtrade/optimize/hyperopt_loss_splinehyperoptloss.py

```diff
@@ -0,0 +1,129 @@
+import logging
+from datetime import datetime
+
+from freqtrade.data.btanalysis import calculate_max_drawdown
+from freqtrade.optimize.hyperopt import IHyperOptLoss
+from pandas import DataFrame
+from scipy import interpolate
+
+logger = logging.getLogger(__name__)
+
+import numpy as np
+
+# HyperOpt Loss function, it is difficult to put the various optimizations into one number.
+# To make it easier, you can specify one table per optimization here. Missing values are interpolated.
+# In this way it is possible to create your own loss curves for each value quite easily.
+
+#
+# this function also removes extrem vales from the results, to optimize on the major mass to avoid
+# single trades/pairs to dominate
+#
+
+class SplineHyperOptLoss(IHyperOptLoss):
+
+    @staticmethod
+    def hyperopt_loss_function(results: DataFrame, trade_count: int,
+                               min_date: datetime, max_date: datetime,
+                               *args, **kwargs) -> float:
+
+        def remove_outliners(x) -> np.array:
+            x = x[x.between(x.quantile(.15), x.quantile(.85))]
+            return x
+
+        def execute_table(table: dict, x: float) -> float:
+            x_points = [float(k) for k in table.keys()]
+            y_points = list(table.values())
+            tck = interpolate.splrep(x_points, y_points, k=2)
+            return float(interpolate.splev(x, tck))
```

Comment:
would it make sense to have a specific test for this method? it's logic seems complex (actually not complex, but easy to break - which usually justifies extra tests).
I can help with tests if that's something you're not familiar with.

to make it testable - it should be moved to class level - maybe with an underscore showing it's a private method).